src/app/stack/stack-details/stack-details.component.ts

```diff
@@ -227,7 +234,104 @@ export class StackDetailsComponent implements OnChanges {
+        length: 0
+    };
+    tokenErrorStatus: boolean = true;
+    tokenDetail: TokenDetailModel = {
+        id: '',
+        status: 'freetier'
+    }
+    tokenAlertsMessage: string;
+
+
+    observableToken: Observable<any>;
+
+    constructor(private stackAnalysisService: StackAnalysesService, private modalService: BsModalService) { }
+
+    async submitToken() {
+        if (!this.tokenErrorStatus) {
+            await this.stackAnalysisService.linkSynkTokenWithUserID(this.getBaseUrl(this.stack), this.uuid, this.token, this.gatewayConfig)
+                .then(res => {
+                    console.log(res.status);
+                })
+                .catch(err => {
+                    console.log(err);
+                });
+
+            console.log('refresh in 5 sec');
+            setTimeout(() => {
+                this.init()
+            }, 5000);
+        }
+
+    }
+
+    async setTokenStatus() {
+
+        if (this.uuid !== null) {
+
+            await this.stackAnalysisService.getTokenStatus(this.getBaseUrl(this.stack), this.uuid, this.gatewayConfig)
+                .then(res => {
+                    this.tokenDetail.id = res.id;
+                    this.tokenDetail.status = res.status.toLowerCase();
+                })
+                .catch(err => {
+                    console.log(err);
+                });
+
+            let resultDetails = this.tokenDetail.status;
+
+            switch (resultDetails.toLowerCase()) {
+                case 'freetier':
+                    this.tokenAlertsMessage = 'Unregistered'
+                    break;
+                case 'registered':
+                    this.tokenAlertsMessage = 'Registered'
+                    break;
+                case 'expired':
+                    this.tokenAlertsMessage = 'Token Expired'
+                    break;
+                default:
+                    break;
+            }
+        } else {
+            return;
+        }
+    }
+
+    openModal(template: TemplateRef<any>) {
+        this.modalRef = this.modalService.show(template);
+    }
+
+    checkToken() {
+        this.tokenerror = {
+            status: false,
+            type: '',
+            length: 0
+        };
+        this.tokenerror.length = this.token.length;
+        if (this.token.length === 0 || this.token == undefined) {
+            this.tokenerror.status = true;
+            this.tokenerror.type = '';
+        } else if (this.token.length !== 36) {
+            this.tokenerror.status = true;
+            this.tokenerror.type = "Token length should be of 36 character";
+        } else if (!/^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/.test(this.token)) {
+            this.tokenerror.status = true;
+            this.tokenerror.type = "Token can only have [a-zA-Z0-9-]";
+        }
+
+        if (this.tokenerror.status) {
+            this.tokenErrorStatus = true;
+        } else {
+            this.tokenErrorStatus = false;
+        }
```

Comment:
Can it be rearranged like below?
```suggestion
        this.tokenErrorStatus = this.tokenerror.status===true;
```

Reply:
Yes. Done